torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -580,46 +576,39 @@ BroadcastOp::BroadcastOp(
     return;
   }
 
-  passkey.ir_container_->registerExpr(exprPasskey(), this);
-
-  // This is a generic check that root dims of a consumer and producer match.
-  // Maybe we shouldn't relegate it to this constructor.
-  const auto c_tv = out_->as<TensorView>();
-  const auto p_tv = in_->as<TensorView>();
-
-  const auto& c_root = c_tv->getRootDomain();
-  const auto& p_root = p_tv->getMaybeRFactorDomain();
-
-  const auto root_p2c =
-      PairwiseRootDomainMap(p_tv, c_tv)
-          .mapProducerToConsumer(p_tv->domain(), c_tv->domain());
-
-  for (auto id : p_root) {
-    if (root_p2c.find(id) == root_p2c.end()) {
+  auto in_tv = in->as<TensorView>();
+  auto out_tv = out->as<TensorView>();
+  auto in_dom = TensorDomain::noReductions(in_tv->getMaybeRFactorDomain());
+  auto& out_dom = out_tv->getRootDomain();
+  TORCH_INTERNAL_ASSERT(
+      is_broadcast_dims_.size() == out_dom.size(),
+      "The dimensions of output tensor and does not match with is_broadcast_dims");
+
+  auto in_size = is_broadcast_dims_.size();
```

Comment:
nit: it was a little difficult to follow here. Probably because at this point `is_broadcast_dims_.size()` isn't the size of input tensor. Also, `in_size--` when `is_broadcast_dims_[i]` is true was a little confusing. I'd say have `int num_new_broadcasts = 0;` here and increment it instead of `in_size--` would make the code easier to follow. The final assertion could be something like `num_new_broadcasts + in_dom.size() == is_braodcast_dims_.size()`.

Reply:
fixed